posts/2017-12-20-who-implements-typeclass.md

```diff
@@ -0,0 +1,565 @@
+---
+layout: post
+title: Who implements the typeclass instance?
+
+meta:
+  nav: blog
+  author: S11001001
+  pygments: true
+---
+
+The typeclass pattern in Scala invites you to place
+implementation-specific knowledge directly in the typeclass instances,
+with the interface defined as the typeclass’s abstract interface.
+
+However, GADTs permit a different organization of code. It is even
+possible to define a typeclass that seems to do nothing at all, yet
+still permits full type-safe typeclass usage.
+
+The possibilities between these two extremes form a design space. If
+you wish to practice ad-hoc polymorphism in Scala, this space is well
+worth exploring.
+
+## A glorified overloader
+
+Refactoring a set of overloads into a typeclass is a fine way to get
+some free flexibility and dynamism, because expressing overloads as a
+typeclass gives you free fixes for common overload problems.
+
+1. Methods calling the overloaded method do not themselves need to be
```

Comment:
This point will become clear later on, but it's a bit difficult to understand here. Maybe add "(see next section)" here?

Reply:
272104c should help a little; thanks.